### PR TITLE
Feature/exam

### PR DESCRIPTION
**Pull request by RobelFekadu**

### CODE REVIEW

File: src/exam/dto/create-exam.dto.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  @IsNumber()
+  readonly price: number;
+
+  @ArrayMinSize(1)
+  readonly tags: string[];
+
+  @IsOptional()
+  @IsArray()
+  readonly samples: string[];
```
Comment (DoniAbeje) [inline src/exam/dto/create-exam.dto.ts]: I think we can remove this, right?

Comment (DoniAbeje) [reply src/exam/dto/create-exam.dto.ts]: I mean, sample questions should be added after the exam is created

File: src/exam/exam-question.service.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      }
+
+      keySet.add(choice.key.toLocaleLowerCase());
+      choiceSet.add(choice.choice.toLocaleLowerCase());
+    }
+
+    // check if the given correctAnswer value(key) is part of the choice
+    if (!correctAnswerKeyFound) {
+      throw new AnswerKeyNotPartOfChoiceException();
+    }
```
Comment (DoniAbeje) [inline src/exam/exam-question.service.ts]: we can put this into a separate method and reuse it for validation

File: src/exam/exam-question.service.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    return examQuestion;
+  }
+
+  async updateExamQuestion(
+    examQuestionId: string,
+    updateExamQuestionDto: UpdateExamQuestionDto,
+  ) {
+    const examQuestion = await this.exists(examQuestionId);
```
Comment (DoniAbeje) [inline src/exam/exam-question.service.ts]: we can use the above validation here

File: src/exam/dto/create-exam.dto.ts (hunk 1 above)
Comment (RobelFekadu) [reply src/exam/dto/create-exam.dto.ts]: are sample questions subset of exam questions?

**Review by DoniAbeje — APPROVED** [review]

**Review by DoniAbeje — APPROVED** [review]

**Pull request closed without merging**